cmd/kubeadm/app/phases/upgrade/staticpods_test.go

```diff
@@ -442,7 +443,7 @@ func TestStaticPodControlPlane(t *testing.T) {
 	for i := range tests {
 		rt := tests[i]
 		t.Run(rt.description, func(t *testing.T) {
-			t.Parallel()
+			pkiutiltesting.Reset()
```

Comment:
wondering, were the tests here and TestRenewCertsByComponent racy or should not run pkiutiltesting based test in parallel (it does have a `lock` for it's functions)?

Reply:
These tests were made parallel in order to speed them up, but the reason they were slow was private key generation, not artificial waits in the tests. Running in parallel didn't help in CPU or random-constrained environments.

When using pkiutil/testing, we want a test to run, asking for private keys, then reset for the next test. Otherwise another test's Reset() could result in already-used private keys being handed out again to a still-running parallel test